core/monitoring/monitoring.py

```diff
@@ -72,5 +105,201 @@ def apply_config(self, config):
+        self._behavior_monitoing_by_status(app_name, "success", request_message_name)
+
+    @silence_it
+    def counter_behavior_fail(self, app_name, request_message_name):
+        # 'Number of Fail replies on messageName'
+        self._behavior_monitoing_by_status(app_name, "fail", request_message_name)
+
+    @silence_it
+    def counter_behavior_misstate(self, app_name, request_message_name):
+        # 'Number of Misstate replies on messageName'
+        self._behavior_monitoing_by_status(app_name, "misstate", request_message_name)
+
+    @silence_it
+    def counter_behavior_timeout(self, app_name, request_message_name):
+        # 'Number of Timeout replies on messageName'
+        self._behavior_monitoing_by_status(app_name, "timeout", request_message_name)
+
+    @silence_it
+    def counter_behavior_expire(self, app_name, request_message_name):
+        # 'Number of expire events on messageName'
+        self._behavior_monitoing_by_status(app_name, "expire", request_message_name)
+
+    def _behavior_monitoing_by_status(self, app_name, status, request_message_name):
+        monitoring_msg = '{}_callback'.format(app_name)
+        c = self._get_or_create_counter(monitoring_msg,
+                                        "Count of incoming callback events with request_message_name",
+                                        ['request_message_name', 'status'])
+
+        c.labels(request_message_name, status).inc()
+
+    @silence_it
+    def counter_host_has_changed(self, app_name):
+        monitoring_msg = '{}_host_has_changed'.format(app_name)
+        c = self._get_or_create_counter(monitoring_msg,
+                                        "Count of host has changed events within one message_id")
+        c.inc()
+
+    @silence_it
+    def counter_mq_long_waiting(self, app_name):
+        monitoring_msg = "{}_mq_long_waiting".format(app_name)
+        c = self._get_or_create_counter(_filter_monitoring_msg(monitoring_msg),
+                                        "(Now - creation_time) is greater than threshold")
+        c.inc()
+
+    @silence_it
+    def sampling_load_time(self, app_name, value):
+        monitoring_msg = "{}_load_time".format(app_name)
+        monitoring.got_histogram_observe(_filter_monitoring_msg(monitoring_msg), value)
+
+    @silence_it
+    def sampling_script_time(self, app_name, value):
+        monitoring_msg = "{}_script_time".format(app_name)
+        monitoring.got_histogram_observe(_filter_monitoring_msg(monitoring_msg), value)
+
+    @silence_it
+    def sampling_save_time(self, app_name, value):
+        monitoring_msg = "{}_save_time".format(app_name)
+        monitoring.got_histogram_observe(_filter_monitoring_msg(monitoring_msg), value)
+
+    @silence_it
+    def sampling_mq_waiting_time(self, app_name, value):
+        monitoring_msg = "{}_mq_waiting_time".format(app_name)
+        monitoring.got_histogram_observe(_filter_monitoring_msg(monitoring_msg), value)
+
+
+class Proxy:
```

Comment:
```suggestion
class ProxyMonitoring:
```
Не кажется ли тебе название Proxy слишком абстрактным?